RevenueCatUI/CustomerCenter/Views/CustomerCenterView.swift

```diff
@@ -94,12 +94,12 @@ private extension CustomerCenterView {
         if viewModel.hasSubscriptions {
             if viewModel.subscriptionsAreFromApple,
                let screen = configuration.screens[.management] {
-                if ignoreAppUpdateWarning || viewModel.appIsLatestVersion {
+                if ignoreAppUpdateWarning || viewModel.appIsLatestVersion || configuration.productId == nil {
                     ManageSubscriptionsView(screen: screen,
                                             customerCenterActionHandler: viewModel.customerCenterActionHandler)
                 } else {
                     AppUpdateWarningView(
-                        onUpdateAppClick: { viewModel.onAppUpdateClick() },
+                        productId: configuration.productId!,
```

Comment:
This is currently safe given the code above, but changes above could render it un-safe. I would recommend turning the else into an else if let statement which can be used to safely unwrap `productId`, and then have an else block that displays maybe content unavailable and logs an error.

Reply:
Yes that's a good point. I was a bit conflicted about this as I didn't want to add an entire view just because the compiler isn't able to figure out that `productId` is actually nil here. 

I ended up inverting the `if` statement, so the `productId` is safely unwrapped in the initial condition. This avoids a useless view and keeps the compiler happy. See https://github.com/RevenueCat/purchases-ios/pull/4199/commits/a55224efbe047f7c197fd58a64da74cf8adef4d2.

Reply:
Yea [smart casts](https://kotlinlang.org/docs/typecasts.html#smart-casts) are amazing, and they [got even better](https://kotlinlang.org/docs/whatsnew20.html#smart-cast-improvements) in Kotlin 2 (which we're not using anywhere yet though).